Master port to OM - Output/Layout

s7design opened the pull request:
Hello,

Hereby is first part of Layout port to OM. LayoutLinkObject.pm and LinkObjectTicket.pm will be ported in next PR.

Regards,
Sanjin

mgruner commented on Kernel/Output/HTML/Layout/Datepicker.pm:
```diff
@@ -36,30 +41,35 @@ Returns a hash of all vacation days defined in the system.
     }
 
+    # get language object
+    my $LanguageObject = $Kernel::OM->Get('Kernel::Language');
```
Here it would be better to keep using $Self->{LanguageObject}. This will stay for now in the LayoutObject as it is used in many places in OTRS.

mgruner commented on Kernel/Output/HTML/Layout/Ticket.pm:
```diff
@@ -37,7 +51,7 @@ sub AgentCustomerViewTable {
     }
     elsif ( ref $Param{Data} eq 'HASH' && !%{ $Param{Data} } ) {
-        return $Self->{LanguageObject}->Translate('none');
+        return $Kernel::OM->Get('Kernel::Language')->Translate('none');
```
Similar here.

s7design commented on Kernel/Output/HTML/Layout/Loader.pm:
```diff
@@ -223,11 +234,14 @@ sub LoaderCreateCustomerCSSCalls {
 
     $Self->{LoaderObject} ||= Kernel::System::Loader->new( %{$Self} );
```
Hi @mgruner ,

Just got a question/confirmation on this. Should we load loader object this way, or use object manager and ObjectParamAdd function to pass whole $Self to it.

Regards,
Sanjin

mgruner replied:
Well the loader takes no constructor arguments, right? So you don't have to pass anything. The ||= is no more needed as the OM keeps the loader copy. You can just use my $LoaderObject = $Kernel::OM->Get() instead.

s7design commented on Kernel/Output/HTML/Layout/Ticket.pm:
```diff
@@ -146,7 +163,7 @@ sub AgentCustomerViewTable {
+                    if ( !$MainObject->Require('Kernel::System::Valid') ) {
                         $Self->FatalDie();
                     }
 
```
Hi @mgruner

This is same question like in Loader.pm (line 235). To keep calling object like it is now, or to use object manager.

Regards,
Sanjin

mgruner replied:
Just switch to OM. This extra handling is not really needed.

s7design commented:
Hi @mgruner 

PR is updated accordingly to your comments.

Regards,
Sanjin

s7design commented:
Hi @mgruner 

I updated the PR as you suggested.
I also changed a little the code.
As you can see I deleted from all modules: 
 * our @ObjectDependencies = (
and added
+our $ObjectManagerDisabled = 1.

These Layout_ modules is not real class, and there will not be needed creating instance of them, or creating objects.

I am not sure about this, but this is my proposal for now, if I am wrong, I will be able to change that.
I don't now how I can test if Loader.pm works well, I tested all modules from Selenium/Output and all of them worked fine
If you have any suggestion please let me know. 

Regards
Zoran

mgruner commented:
Merged, thank you!
